### ISSUE TITLE
LinkedIn wants different characters escaped

### ISSUE THREAD

**Issue by pj4533** [AFNetworking/AFOAuth1Client#24]
I was having trouble in my AFLinkedInOAuth1Client class, and tracked down the issue to the escaping function in AFOAuth1Client.   I changed it to this to get it working:

``` objc
static NSString * AFPercentEscapedQueryStringPairMemberFromStringWithEncoding(NSString *string, NSStringEncoding encoding) {
    static NSString * const kAFCharactersToBeEscaped = @":/?&=;+!@#$(),";
    static NSString * const kAFCharactersToLeaveUnescaped = @"[].~";
//    static NSString * const kAFCharactersToBeEscaped = @":/?&=;+!@#$()~";
//    static NSString * const kAFCharactersToLeaveUnescaped = @"[].";

    return (__bridge_transfer  NSString *)CFURLCreateStringByAddingPercentEscapes(kCFAllocatorDefault, (__bridge CFStringRef)string, (__bridge CFStringRef)kAFCharactersToLeaveUnescaped, (__bridge CFStringRef)kAFCharactersToBeEscaped, CFStringConvertNSStringEncodingToEncoding(encoding));
}
```

Ideally, I'd like to be able to override the escapable characters in my subclass.  Does it make sense to move this static function to something that could be easily overridden?  Or maybe make the kAFCharactersToBeEscape/LeaveUnescaped  as properties on the client?

**Comment by notapplicableio** [comment]
How's this looking? Any movement?

**Comment by dlackty** [comment]
@pj4533 @trustyfrog This should be fixed in AFNetworking/AFNetworking@953b862

**Comment by mattt** [comment]
This has been addressed in AFNetworking, as well as 5c59bcd6b4415bc3a46f1eb2259f1da35043d0c0 in this project. AFNetworking 2.0, thankfully, will use Apple-provided URL string encoding methods, and provide a first-class interface to customizing the parameter encoding to your requirements.
